utils/pckgen

```diff
@@ -0,0 +1,454 @@
+        info "WARNING: Cleanup failed..."
+    fi
+}
+
+check_options() {
+    if [ ! "$SOURCES_DIR" ]; then
+        invalid_usage "No mandatory SOURCES_PATH provided"
+    elif [[ $(head -n 1 "$SOURCES_DIR/README.md" 2> /dev/null) != *Open*CAS*Linux* ]]; then
+        invalid_usage "'$SOURCES_DIR' does not point to the root directory of CAS sources"
+    elif [ ! "${!GENERATE_*}" ] && [ ! $VERSION_ONLY ]; then
+        invalid_usage "Nothing to do - no command provided"
+    fi
+}
+
+check_version() {
+    cd "$TEMP_DIR"
+    sh "$CAS_VERSION_GEN" > /dev/null
+    if [ ! -f "$CAS_VERSION_GEN_FILE" ]; then
+        error "Couldn't create CAS_VERSION file"
+    fi
+    while read -r line; do
+        case "$line" in
+            CAS_VERSION_MAIN*) CAS_VERSION_MAIN="${line##*=}";;
+            CAS_VERSION_MAJOR*) CAS_VERSION_MAJOR="${line##*=}";;
+            CAS_VERSION_MINOR*) CAS_VERSION_MINOR="${line##*=}";;
+        esac
+    done < "$CAS_VERSION_GEN_FILE"
+    rm "$CAS_VERSION_GEN_FILE"
+    cd - > /dev/null
+
+    VERSION_SHORT="${CAS_VERSION_MAIN}.${CAS_VERSION_MAJOR}"
+    if [ $CAS_VERSION_MINOR -ne 0 ]; then
+        VERSION_SHORT+=".${CAS_VERSION_MINOR}"
+    fi
+    if [[ "$VERSION_SHORT" < "$SUPPORTED_FROM_VERSION" ]]; then
+        echo "Sorry... this version of $NAME ($VERSION_SHORT) is not supported"\
+             "by $THIS. Use $NAME >= $SUPPORTED_FROM_VERSION"
+        exit 1
+    fi
+
+    for SUBMOD in ${SUBMODULES[@]}; do
+        if [[ ! $(ls -A "$SOURCES_DIR/$SUBMOD") ]]; then
+            local SUBMODULES_MISSING+="'$SUBMOD'\n"
+        fi
+    done
+    if [ "$SUBMODULES_MISSING" ]; then
+        error "There are missing submodules:\n${SUBMODULES_MISSING}\nUpdate submodules and try again!"
+    fi
+
+    if [ -d "$SOURCES_DIR/.git" ]; then
+        CAS_VERSION_BUILD=$(printf %04d $(git -C "$SOURCES_DIR" log --merges --oneline | wc -l))
+        LAST_COMMIT_HASH=$(git -C "$SOURCES_DIR" log -1 --pretty=format:%H)
+        LAST_COMMIT_HASH_ABBR=$(git -C "$SOURCES_DIR" log -1 --pretty=format:%h)
+        LAST_COMMIT_DATE=$(git -C "$SOURCES_DIR" log -1 --pretty=format:%ci |\
+                           sed "s/ /T/" | sed "s/ //" | sed "s/00$/:00/")
+        LAST_COMMIT_TIMESTAMP=$(git -C "$SOURCES_DIR" log -1 --pretty=format:%ct)
+        for SUBMOD in ${SUBMODULES[@]}; do
+            LAST_SUB_COMMIT_HASHES+=($(git -C "$SOURCES_DIR/$SUBMOD" log -1 --pretty=format:%H))
+            LAST_SUB_COMMIT_HASHES_ABBR+=($(git -C "$SOURCES_DIR/$SUBMOD" log -1 --pretty=format:%h))
+        done
+    elif [ -f "$VERSION_FILE" ]; then
+        source "$VERSION_FILE" &> /dev/null
+    else
+        error "Couldn't obtain CAS version - no git tree nor version file within given sources"
+    fi
+
+    VERSION_LONG="${CAS_VERSION_MAIN}.${CAS_VERSION_MAJOR}.${CAS_VERSION_MINOR}.${CAS_VERSION_BUILD}"
+
+    if [ $(git -C "$SOURCES_DIR" tag --points-at HEAD) ]; then
+        VERSION="$VERSION_SHORT"
+        RELEASE="release"
```

Comment:
My understanding is that the resulting "release" package will be identified as a lower version than preceding "master" packages. I believe this is not desired. It would be better to add a full version number in release package. @robertbaldyga do you agree?

Reply:
Yep. Additionally one of main objectives was to have one unified version format, which IMO should be `VERSION_LONG`. Thus we want to use it also for release builds.